air/src/execution_step/execution_context/stream_maps_variables/stream_map_key.rs

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2023 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use crate::CatchableError;
+use crate::ExecutionError;
+use crate::JValue;
+use crate::StreamMapError::UnsupportedMapKeyType;
+
+use serde_json::Number;
+
+pub(crate) enum StreamMapKey<'i> {
+    Str(&'i str),
+    Number(Number),
+}
+
+impl<'i> StreamMapKey<'i> {
+    pub(crate) fn from_value(value: &'i JValue, map_name: &str) -> Result<Self, ExecutionError> {
+        match value {
+            JValue::String(s) => Ok(StreamMapKey::Str(s.as_str())),
+            JValue::Number(n) if n.is_i64() || n.is_u64() => Ok(StreamMapKey::Number(n.clone())),
```

Comment:
To improve code readability, it's better to create a new function for this if with a self-explanatory name, smth like that 
```rust
fn is_allowed_number_type(n: ...) -> bool {
   n.is_i64() || n.is_u64()
}
```

Reply:
I doubt that `n.clone()` is needed here, because it should implement the `Copy` trait.